### ISSUE TITLE
XML serializer can't handle recursive enumeration types

### ISSUE THREAD

**Issue by SzymonPobiega** [Particular/NServiceBus#4715]
Types that implement `IEnumerable` of themselves cause XML serializer to boom with `StackOverflowException`. An example is `JToken`:

```
Assert.IsTrue(typeof(IEnumerable<JToken>).IsAssignableFrom(typeof(JToken)));
```

The problematic line is 54 in `XmlSerializerCache`.

**Comment by timbussmann** [comment]
@SzymonPobiega as you assigned this yourself, are you actively doing something related to this issue?

**Comment by SzymonPobiega** [comment]
@timbussmann Nope. I've assigned it to not lose it. Feel free to un-assign me.

**Comment by boblangley** [comment]
We have triaged this issue and will be considering it for inclusion in one of the upcoming maintenance releases for this package.
